src/field/launch_application.c

```diff
@@ -0,0 +1,1400 @@
+    args->unk4 = sub_0203EE54(saveData);
+    _LaunchPokeathlonCourseRecordApp(fieldSystem, args);
+    return args;
+}
+
+static void _LaunchPokeathlonMedalsApp(FieldSystem *fieldSystem, PokeathlonMedalsArgs *args) {
+    FieldSystem_LaunchApplication(fieldSystem, &sOvyTemplate_PokeathlonMedals, args);
+}
+
+PokeathlonMedalsArgs *LaunchPokeathlonMedalsApp(FieldSystem *fieldSystem) {
+    PokeathlonMedalsArgs *args = AllocFromHeapAtEnd(HEAP_ID_FIELD, sizeof(PokeathlonMedalsArgs));
+    SaveData *saveData = FieldSystem_GetSaveData(fieldSystem);
+    POKEATHLON_SAV *pokeathlonSave = Save_Pokeathlon_Get(saveData);
+    Pokedex *pokedex = Save_Pokedex_Get(saveData);
+    args->pokedex = pokedex;
+    args->unk4 = sub_02031978(pokeathlonSave);
+    args->natDexEnabled = Pokedex_GetNatDexFlag(pokedex);
+    args->unkC = sub_0203EE54(saveData);
+    _LaunchPokeathlonMedalsApp(fieldSystem, args);
+    return args;
+}
+
+static void _LaunchPokeathlonEventRecordApp(FieldSystem *fieldSystem, PokeathlonEventRecordArgs *args) {
+    FieldSystem_LaunchApplication(fieldSystem, &sOvyTemplate_PokeathlonEventRecord, args);
+}
+
+PokeathlonEventRecordArgs *LaunchPokeathlonEventRecordApp(FieldSystem *fieldSystem) {
+    PokeathlonEventRecordArgs *args = AllocFromHeapAtEnd(HEAP_ID_FIELD, sizeof(PokeathlonEventRecordArgs));
+    SaveData *saveData = FieldSystem_GetSaveData(fieldSystem);
+    POKEATHLON_SAV *pokeathlonSave = Save_Pokeathlon_Get(saveData);
+    args->unk0 = sub_0203197C(pokeathlonSave);
+    args->unk4 = sub_02031984(pokeathlonSave);
+    args->profile = Save_PlayerData_GetProfileAddr(saveData);
+    args->unk10 = sub_0203EE54(saveData);
+    args->unk8 = sub_02031990(pokeathlonSave);
+    _LaunchPokeathlonEventRecordApp(fieldSystem, args);
+    return args;
+}
+
+static void _LaunchPokeathlonUnkApp(FieldSystem *fieldSystem, Unk0203EFA0 *args) {
+    FieldSystem_LaunchApplication(fieldSystem, &sOvyTemplate_PokeathlonUnkApp, args);
+}
+
+Unk0203EFA0 *LaunchPokeathlonUnkApp(FieldSystem *fieldSystem) {
+    Unk0203EFA0 *args = AllocFromHeapAtEnd(HEAP_ID_3, sizeof(Unk0203EFA0));
+    SaveData *saveData = FieldSystem_GetSaveData(fieldSystem);
+    PlayerProfile *profile = Save_PlayerData_GetProfileAddr(saveData);
+    args->pokeathlon = Save_Pokeathlon_Get(saveData);
+    args->profile = profile;
+    _LaunchPokeathlonUnkApp(fieldSystem, args);
+    return args;
+}
+
+static void sub_0203EFD4(FieldSystem *fieldSystem, UseMailArgs *args) {
+    args->unk1C = &fieldSystem->unk_10C;
+    FieldSystem_LaunchApplication(fieldSystem, &_020FA434, args);
+}
+
+UseMailArgs *sub_0203EFEC(FieldSystem *fieldSystem, u16 a1, u8 partyIdx, u8 a3) {
+    UseMailArgs *args = sub_02090E68(FieldSystem_GetSaveData(fieldSystem), a1, partyIdx, a3, HEAP_ID_FIELD);
+    sub_0203EFD4(fieldSystem, args);
+    return args;
+}
+
+UseMailArgs *CreateUseMailWork(FieldSystem *fieldSystem, int kind, u8 mailId, HeapID heapId) {
+    UseMailArgs *args;
+
+    if (kind == 3) {
```

Comment:
what is kind? const?

Reply:
Renamed to `n` -- that's what `sub_02090EC0()` calls it.

Reply:
n isn't more descriptive than kind, it's much less, if it's not known that's ok, but if it is I'd prefer to have it documented